submodules/TelegramCore/TelegramCore/CirclePeers.swift

```diff
@@ -16,31 +16,40 @@ struct ApiCircle {
     let peers: [PeerId]
 }
 
-func fetchCircles(postbox: Postbox) -> Signal<Void, NoError> {
+func fetchCircles(postbox: Postbox, userId: PeerId) -> Signal<Void, NoError> {
     return getCirclesSettings(postbox: postbox)
     |> mapToSignal { settings -> Signal<[ApiCircle],NoError> in
         if let token = settings?.token {
             return Signal<[ApiCircle], NoError> { subscriber in
-                
                 let urlString = Circles.baseApiUrl
                 Alamofire.request(
                     urlString,
                     headers: ["Authorization": token]
                 ).responseJSON { response in
                     if let result = response.result.value {
                         let json = SwiftyJSON.JSON(result)
-                        let circles = json["circles"].arrayValue.map { circle in
-                            return ApiCircle(
-                                id: PeerGroupId(rawValue: circle["id"].int32Value),
-                                name: circle["name"].stringValue,
-                                peers: circle["peers"].arrayValue.map { idObject in
-                                    let apiId = idObject.int32Value
-                                    if  apiId > 0 {
-                                        return PeerId(namespace: Namespaces.Peer.CloudUser, id: apiId)
+                        let circles:[ApiCircle] = json["circles"].arrayValue.map { circle in
+                            let idArray = circle["peers"].arrayValue + circle["members"].arrayValue
+                            let peers:[PeerId] = idArray
+                            .filter { $0.int32Value != userId.id }
+                            .map { idObject in
+                                let apiId = idObject.int64Value
+                                if  apiId > 0 {
+                                    return PeerId(namespace: Namespaces.Peer.CloudUser, id: idObject.int32Value)
+                                } else {
+                                    let trillion:Int64 = 1000000000000
```

Comment:
Why a hardcode? We should never do anything like that. This should be a const already defined in Telegram. If their backend adds that, this should live on our backend. @michaelenco @youaresofunny